src/primaite/agents/llm/prompting.py

```diff
@@ -0,0 +1,149 @@
+        return NodeAction.from_text(
+            env=env,
+            node_name=self.node_name,
+            node_property=self.node_property,
+            property_action=self.property_action,
+            service_name=self.service_name,
+        )
+
+
+class AgentReasoningNodeSelection(BaseModel):
+    reasoning: str = Field(..., description="Communication of observational understanding of any actions required.")
+    node_name: str = Field(
+        "NONE", description="Selection of a node of highest priorityfor which to perform an action on."
+    )
+
+
+SYSTEM_MSG = "You are a cybersecurity network defence agent. Your mission is to protect a network against offensive attacks. You have a limited view of the network environment, known as an observation space. The network is made up of nodes (e.g. computers, switches) and links between some of the nodes, through which information is transmitted using standard protocols. The current link traffic is displayed as a percentage of its total bandwidth, for each possible protocol. An attacker is trying to compromise the network, by either directly incapacitating the nodes HARDWARE, SOFTWARE or FILE_SYSTEM states, or indirectly overwhelming the nodes through Denial-of-Service type attacks.\n"
+
+REASON_ACTION_SPACE_NODE_SELECT = """
+This is the initial configuration of the network:
+{network_connectivity_desc}
+
+Initial
+{initial_obs_view_full}
+
+This is the history of offensive action observations and defensive actions you have taken at each step. If nothing happened at a specific step, it is omitted from the history:
+{obs_act_history}
+
+Here is an overview of the current observation space:
+{current_obs_view_full}
+Changes that have occured since last observation are:
+{current_obs_diff}
+
+As an agent, you are able to influence the state of this node by switching it on or off, resetting it, patching software or patching any of its services.
+
+Please think about the network configuration and the state of each node. Think about which nodes are most vulnerable to attack and which node requires action the most in order to stop the attack and prevent further spread. Provide your reasoning statement and select a node by name to perform a defensive action on. If no action is required because all is well, you can simply say 'NONE', but always provide a reasoning statement.
+
+If action is not required at the moment, set NONE as the node_name, but always reason over the state of the network.
+Note that actions are expensive and can negatively impact the environment if used improperly. For instance, a server which is turned off cannot receive requests from the users and will decrease the reward.  
+
+If choosing to take an action, you must select a node by name from the following list: {node_names}.
+
+
+For your information, the following actions are available for selection later. Always take note of any action constraints outlined in the description provided.
+
+## HARDWARE Actions:
```

Comment:
maybe we could store this action description in a separate variable so that we can reuse it in the prompts